@sengineer0 I saw an error from a test of `quick_index` feature:

```
Jun 25 20:49:42 su06 python[1658027]: INFO:index_diseases_20220625_p8wlbepn:#1: 6438 documents.
Jun 25 20:49:44 su06 python[1657068]: INFO:tornado.access:304 GET /command/23835 (172.29.200.97) 0.96ms
Jun 25 20:49:44 su06 python[1657068]: INFO:tornado.access:200 GET /job_manager (172.29.200.97) 13.54ms
Jun 25 20:49:46 su06 python[1658027]: ERROR:asyncio:Task was destroyed but it is pending!
Jun 25 20:49:46 su06 python[1658027]: task: <Task pending name='Task-172' coro=<WebSocketProtocol13.write_message.<locals>.wrapper() running at /opt/home/pending/venv/lib/python3.8/site-packages/tornado/websocket.py:1100>>
Jun 25 20:49:46 su06 python[1658027]: /usr/lib/python3.8/asyncio/base_events.py:641: RuntimeWarning: coroutine 'WebSocketProtocol13.write_message.<locals>.wrapper' was never awaited
Jun 25 20:49:46 su06 python[1658027]:   self._ready.clear()
Jun 25 20:49:46 su06 python[1658027]: RuntimeWarning: Enable tracemalloc to get the object allocation traceback
Jun 25 20:49:46 su06 python[1658027]: ERROR:asyncio:Task was destroyed but it is pending!
Jun 25 20:49:46 su06 python[1658027]: task: <Task pending name='Task-180' coro=<WebSocketProtocol13.write_message.<locals>.wrapper() running at /opt/home/pending/venv/lib/python3.8/site-packages/tornado/websocket.py:1100>>
Jun 25 20:49:47 su06 python[1657068]: INFO:tornado.access:200 GET /job_manager (172.29.200.97) 12.24ms
Jun 25 20:49:49 su06 python[1657068]: INFO:tornado.access:304 GET /command/23835 (172.29.200.97) 0.94ms
Jun 25 20:49:50 su06 python[1657068]: INFO:tornado.access:200 GET /job_manager (172.29.200.97) 13.54ms
Jun 25 20:49:53 su06 python[1657068]: INFO:tornado.access:200 GET /job_manager (172.29.200.97) 12.91ms
Jun 25 20:49:54 su06 python[1657068]: INFO:tornado.access:304 GET /command/23835 (172.29.200.97) 0.93ms
Jun 25 20:49:54 su06 python[1658027]: DEBUG:hub:Remove PID file '/data/pending/run/1658027_g7dh1eIS.pickle'
Jun 25 20:49:54 su06 python[1657068]: ERROR:index_diseases_20220625_p8wlbepn:(5703, 6438)
Jun 25 20:49:54 su06 python[1657068]: Traceback (most recent call last):
Jun 25 20:49:54 su06 python[1657068]:   File "/opt/home/pending/venv/src/biothings/biothings/hub/dataindex/indexer.py", line 376, in index
Jun 25 20:49:54 su06 python[1657068]:     dx = await step.execute(job_manager, **kwargs)
Jun 25 20:49:54 su06 python[1657068]:   File "/opt/home/pending/venv/src/biothings/biothings/hub/dataindex/indexer.py", line 242, in execute
Jun 25 20:49:54 su06 python[1657068]:     return (await coro)
Jun 25 20:49:54 su06 python[1657068]:   File "/opt/home/pending/venv/src/biothings/biothings/hub/dataindex/indexer.py", line 517, in do_index
Jun 25 20:49:54 su06 python[1657068]:     schedule.completed()
Jun 25 20:49:54 su06 python[1657068]:   File "/opt/home/pending/venv/src/biothings/biothings/hub/dataindex/indexer_schedule.py", line 40, in completed
Jun 25 20:49:54 su06 python[1657068]:     raise ValueError(self.finished, self.total)
Jun 25 20:49:54 su06 python[1657068]: ValueError: (5703, 6438)
Jun 25 20:49:54 su06 python[1657068]: DEBUG:indexmanager:<Task finished name='Task-960' coro=<Indexer.index() done, defined at /opt/home/pending/venv/src/biothings/biothings/hub/dataindex/indexer.py:339> exception=ValueError(5703, 6438)>
Jun 25 20:49:55 su06 python[1657068]: INFO:tornado.access:200 GET /command/23835 (172.29.200.97) 1.26ms

```

--- a/biothings/hub/__init__.py
+++ b/biothings/hub/__init__.py
@@ -1550,6 +1550,10 @@ class HubServer(object):
         index_name = index_name.lower()
 
         async def do():
+            extra_index_settings = kwargs.pop("extra_index_settings", '{}')
+            extra_index_settings = json.loads(extra_index_settings)
+            extra_index_settings["num_shards"] = int(kwargs.pop("num_shards", 1))
+            extra_index_settings["num_replicas"] = int(kwargs.pop("num_replicas", 1))
             try:
                 # create a temporary build configuration:
                 builder_class = None
@@ -1562,6 +1566,7 @@ class HubServer(object):
                     doc_type=doc_type,
                     sources=[datasource_name],
                     builder_class=builder_class,
+                    params=extra_index_settings,
                 )
 
                 # create a temporary build

--- a/biothings/hub/__init__.py
+++ b/biothings/hub/__init__.py
@@ -1553,7 +1553,7 @@ class HubServer(object):
             extra_index_settings = kwargs.pop("extra_index_settings", '{}')
             extra_index_settings = json.loads(extra_index_settings)
             extra_index_settings["num_shards"] = int(kwargs.pop("num_shards", 1))
-            extra_index_settings["num_replicas"] = int(kwargs.pop("num_replicas", 1))
+            extra_index_settings["num_replicas"] = int(kwargs.pop("num_replicas", 0))
             try:
                 # create a temporary build configuration:
                 builder_class = None

--- a/biothings/hub/dataindex/indexer.py
+++ b/biothings/hub/dataindex/indexer.py
@@ -490,7 +490,7 @@ class Indexer():
             if error:
                 for job in jobs:
                     if not job.done():
-                        job.cancel()
+                        await job.cancel()
                 raise error
 
             self.logger.info(schedule)

